### ISSUE TITLE
Change language in markdown code fences

### ISSUE THREAD

**Issue by gandalfsaxe** [Jason-Rev/vscode-spell-checker#242]
Great extension! However I often make code fences in markdown, resulting in LOTS of false positive when vscode-spell-checker checks the code fence as if it was markdown code. Example:

````markdown
# Some title
Bla bla this is markdown and all is good.

```latex
\usepackage[utf8]{inputenc}
```
````
which would result in warning on `usepackage` and `inputenc`. Here is a screenshot that shows the issue:

![image](https://user-images.githubusercontent.com/12127864/42728831-a846238a-87c4-11e8-92dd-cd9506424915.png)

The best solution would be to dynamically switch to the code fence language locally within the code fence.

The next-best solution would be to just ignore all content without code fences.

**Comment by gandalfsaxe** [comment]
As a workaround, can I manually enable the latex dictionary for a specific file?

**Comment by Jason3S** [comment]
It is possible to ignore those sections:

For all file types:
```
    "cSpell.ignoreRegExpList": [
         "/^```(?:.|\\s)+?^```/mig"
    ],
```

To have it apply only to markdown:
```
"cSpell.languageSettings": [
        {
            "languageId": "markdown",
            "ignoreRegExpList": [
                "/^```(?:.|\\s)+?^```/mig"
             ]
        }
    ]
```

**Comment by GandalfSaxe** [comment]
Thanks, can confirm it works to ignore content in code fences 👍

1. Any hope for locally applying dictionary in code fence according to code fence language?
2. Any way of saying "Enable latex dict when editing markdown files"?

**Comment by GandalfSaxe** [comment]
For inspiration, prettier.io actually implemented detection of language in code fences: https://prettier.io/blog/2018/04/11/1.12.0.html#support-fenced-codeblock-lang-followed-by-attributes-4153-https-githubcom-prettier-prettier-pull-4153-by-ikatyang-https-githubcom-ikatyang

**Comment by noraj** [comment]
Wanted behavior for markdown:

+ code block: follow the language declared
+ code block: if not language is declared (so it is plaintext), disable spellcheck by default or add an option to do it
+ inline code:  disable spellcheck by default or add an option to do it

Else it generate too much false positive:

Plaintext code block:

![image](https://user-images.githubusercontent.com/16578570/51087694-dfbf6700-1756-11e9-9f8e-71c2b4eba496.png)

![image](https://user-images.githubusercontent.com/16578570/51087712-02518000-1757-11e9-9fd0-1c44347a4650.png)


Inline code: 

![image](https://user-images.githubusercontent.com/16578570/51087705-eea61980-1756-11e9-97eb-d67e469539a3.png)

**Comment by peixinshen** [comment]
With regard to enable LaTeX dictionary in the markdown file, my workaround is to change the global setting in `settings.json`,
```
"cSpell.languageSettings": [
        {"languageId": "markdown",  "dictionaries": ["latex"]}
    ]
```

**Comment by gauss256** [comment]
I've posted a solution here: https://github.com/streetsidesoftware/vscode-spell-checker/issues/357?_pjax=%23repo-content-pjax-container#issuecomment-898993398

It doesn't dynamically switch to the code fence language locally within the code fence, but it excludes spell checking in code blocks, which takes care of all the false positives in them.

**Comment by QuintinWillison** [comment]
Thanks, @Jason3S, [your solution](https://github.com/streetsidesoftware/vscode-spell-checker/issues/242#issuecomment-406789352) was very helpful... I ended up tweaking it a little, in order to support situations where the [fenced code block](https://github.github.com/gfm/#fenced-code-blocks) is indented (e.g. when part of a list). Here in case it's of help to anybody:

```json
"ignoreRegExpList": [
  "/^[ \t]*```(?:.|\\s)+?^[ \t]*```/mig"
]
```

**Comment by Jason3S** [comment]
@QuintinWillison,

Thank you. There is a more universal solution that will also works with nested code blocks.

Given:

`````md
  # Example Title
  
  ## Example Code Bock

  ````md
  ```json
  { "value": 42 }
  ```
  ````
`````
